### PR TITLE
fixes Traceback generated by nxos_igmp module

### LINKED ISSUE #27488: (P1)nxos_igmp: module fails with traceback
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

- nxos_igmp

##### ANSIBLE VERSION

```
ansible 2.4.0 (devel b93b9e68d7) last updated 2017/07/26 14:19:36 (GMT -400)
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/agents-ci/ansible/lib/ansible
  executable location = /root/agents-ci/ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### OS / ENVIRONMENT

Ubuntu 14.04

##### SUMMARY

Traceback seen while applying playbook

##### STEPS TO REPRODUCE

```yaml
---
- hosts: nxos
  connection: local
  gather_facts: no

  tasks:
  - name: "Configure igmp with default values"
    nxos_igmp:
      state: default
      provider: "{{ cli }}"
```

##### EXPECTED RESULTS

No tracebacks.

##### ACTUAL RESULTS

```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_D0RHMz/ansible_module_nxos_igmp.py", line 257, in <module>
    main()
  File "/tmp/ansible_D0RHMz/ansible_module_nxos_igmp.py", line 244, in main
    result['connected'] = module.connected   
AttributeError: 'AnsibleModule' object has no attribute 'connected'
FAILED! => {
    "changed": false, 
    "failed": true, 
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_D0RHMz/ansible_module_nxos_igmp.py\", line 257, in <module>\n    main()\n  File \"/tmp/ansible_D0RHMz/ansible_module_nxos_igmp.py\", line 244, in main\n    result['connected'] = module.connected\nAttributeError: 'AnsibleModule' object has no attribute 'connected'\n",  
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "rc": 0
}
```

### PR DESCRIPTION
**Pull request by privateip**
fixes #27488

### CODE REVIEW

**Review by Qalthos — CHANGES_REQUESTED** [review]
This will still fail as candidate needs to be turned into a list of commands (via `.items_text()`) to successfully pass through `load_config()`. Additionally, the response of `load_config()` should no longer be pushed into `result` as it doesn't return anything (notwithstanding my PR, which instead makes it return something we don't want there).

Also, seeing result return a `'commands'` key and updating the docs so it matches most of the rest of the modules would be helpful, even if `'updates'` sticks around.

**Comment by privateip** [comment]
ready_for_review

File: lib/ansible/modules/network/nxos/nxos_igmp.py [hunk 1]
```diff
@@ -206,53 +131,43 @@ def main():
...
-            module.params['enforce_rtr_alert'] is not None)):
-        module.fail_json(msg='When state=default other params have no effect.')
+    current = get_current(module)
+    desired = get_desired(module)
 
-    args =  [
-        "flush_routes",
-        "enforce_rtr_alert",
-    ]
+    import q
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_igmp.py]: Yay for `q`, but might want to take it out.

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/nxos/nxos_igmp.py (hunk 1 above)
Comment (privateip) [reply lib/ansible/modules/network/nxos/nxos_igmp.py]: doh!.. updated

**Pull request merged**